modules/vba_documents/app/serializers/vba_documents/upload_serializer.rb

```diff
@@ -18,8 +18,18 @@ class UploadSerializer
       detail.length > MAX_DETAIL_DISPLAY_LENGTH ? "#{detail[0..MAX_DETAIL_DISPLAY_LENGTH - 1]}..." : detail
     end
 
-    attribute :location do |object, params|
-      object.get_location if params[:render_location]
+    attribute :final_status, if: proc { |_, _|
+      # The final_status will be serialized only if the :vba_documents_final_status_field flag is enabled
+      Flipper.enabled?(:vba_documents_final_status_field)
+    } do |object, _|
+      object.in_final_status?
+    end
+
+    attribute :location, if: proc { |_, params|
+      # The location will be serialized only if the :render_location key of params is true
+      params[:render_location] == true
+    } do |object, _|
+      object.get_location
```

Comment:
This fixes a bug introduced in #17348 where the `location` would be included in the response even if `render_location` wasn't true (the value would be `null`). Instead, the key should be excluded entirely.